src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.QueryParameters.cs

```diff
@@ -109,4 +109,47 @@ public async Task MapAction_MultipleStringParam_StringReturn()
         await VerifyResponseBodyAsync(httpContext, "Hello world!");
         await VerifyAgainstBaselineUsingFile(compilation);
     }
+
+    public static object[][] MapAction_ExplicitQueryParam_NameTest_Data
+    {
+        get
+        {
+
+            return new[]
+            {
+                new object[] { "name", "name" },
+                new object[] { "_", "_" },
+                new object[] { "123", "123" },
+                new object[] { "💩", "💩" },
+                new object[] { "\r", "\\r" },
+                new object[] { "\x00E7" , "\x00E7" },
+                new object[] { "!!" , "!!" },
+                new object[] { "\\" , "\\\\" },
+                new object[] { "\'" , "\'" },
+            };
+        }
+    }
+
+    [Theory]
+    [MemberData(nameof(MapAction_ExplicitQueryParam_NameTest_Data))]
+    public async Task MapAction_ExplicitQueryParam_NameTest(string name, string lookupName)
```

Comment:
I also added route tests: [dotnet/aspnetcore@`jamesnk/generator-escaping?expand=1`#diff-00af5042a9](https://github.com/dotnet/aspnetcore/compare/jamesnk/generator-escaping?expand=1#diff-00af5042a936504ada7633cb1864c9b40225be2afadaea5abae224753e5cce34R155)

Reply:
Yep just copied them over in an incoming commit.